File: leetcode/Arrays/leetcode-0055/55.java
class Solution {
    public boolean canJump(int[] nums) {
        if (nums == null || nums.length == 0) {
            return true;
        }
        int reach = 0;
        for (int i = 0; i < nums.length; i++) {
            if (reach >= nums.length - 1) {
                return true;
            } else if (reach < i) {
                return false;
            }
            reach = Math.max(reach, i + nums[i]);
        }
        return false;
    }
}

File: leetcode/Arrays/leetcode-0055/55.go
package arrays

func canJump(nums []int) bool {
	if len(nums) == 0 {
		return true
	}
	reach := 0
	for i := 0; i < len(nums); i++ {
		if reach >= len(nums)-1 {
			return true
		} else if reach < i {
			return false
		}
		if i+nums[i] > reach {
			reach = i + nums[i]
		}
	}
	return false
}
